### ISSUE TITLE
Make error with anaconda

### ISSUE THREAD

**Issue by Dinghow** [stevenlovegrove/Pangolin#488]
First, to find my python2.7 interpreter in anaconda, I use this command
`cmake -DPYBIND11_PYTHON_VERSION=2.7 ..`

Then I met this problem
```
Scanning dependencies of target VideoViewer
[ 79%] Building CXX object tools/VideoViewer/CMakeFiles/VideoViewer.dir/main.cpp.o
[ 80%] Linking CXX executable VideoViewer
//home/goolglecamp/anaconda2/lib/libpng16.so.16: undefined reference to `inflateValidate@ZLIB_1.2.9'
collect2: error: ld returned 1 exit status
tools/VideoViewer/CMakeFiles/VideoViewer.dir/build.make:102: recipe for target 'tools/VideoViewer/VideoViewer' failed
make[2]: *** [tools/VideoViewer/VideoViewer] Error 1
CMakeFiles/Makefile2:348: recipe for target 'tools/VideoViewer/CMakeFiles/VideoViewer.dir/all' failed
make[1]: *** [tools/VideoViewer/CMakeFiles/VideoViewer.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

I've seen [issue#285](https://github.com/stevenlovegrove/Pangolin/issues/285), but it didn't make sense.

**Comment by BurakHocaoglu** [comment]
Hello there,

I don't know if it will work, but I have faced a similar issue when building while anaconda was active, where in my case it was:
```
../../src/libpangolin.so: undefined reference to `png_set_longjmp_fn'
collect2: error: ld returned 1 exit status
tools/VideoViewer/CMakeFiles/VideoViewer.dir/build.make:106: recipe for target 'tools/VideoViewer/VideoViewer' failed
make[2]: *** [tools/VideoViewer/VideoViewer] Error 1
CMakeFiles/Makefile2:294: recipe for target 'tools/VideoViewer/CMakeFiles/VideoViewer.dir/all' failed
make[1]: *** [tools/VideoViewer/CMakeFiles/VideoViewer.dir/all] Error 2
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2
```

When I run the first cmake, I got many warnings of the following type:
```
CMake Warning at tools/VideoViewer/CMakeLists.txt:5 (add_executable):
  Cannot generate a safe runtime search path for target VideoViewer because
  files in some directories may conflict with libraries in implicit
  directories:

    runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/kovan-fl/anaconda3/lib
    runtime library [libtiff.so.5] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/kovan-fl/anaconda3/lib
    runtime library [liblz4.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
      /home/kovan-fl/anaconda3/lib

  Some of these libraries may not be found correctly.
```

Then, I deleted the old build and tried to build without anaconda, and it worked. Also, the dependencies are installed via ```sudo apt-get install```, not with ```conda install```; therefore, I believe, under normal circumstances, you should not see an error about dependencies.

Regards,

Burak

**Comment by Dinghow** [comment]
> Hello there,
> 
> I don't know if it will work, but I have faced a similar issue when building while anaconda (Python 3.7.3) was active, where in my case it was:
> 
> ```
> ../../src/libpangolin.so: undefined reference to `png_set_longjmp_fn'
> collect2: error: ld returned 1 exit status
> tools/VideoViewer/CMakeFiles/VideoViewer.dir/build.make:106: recipe for target 'tools/VideoViewer/VideoViewer' failed
> make[2]: *** [tools/VideoViewer/VideoViewer] Error 1
> CMakeFiles/Makefile2:294: recipe for target 'tools/VideoViewer/CMakeFiles/VideoViewer.dir/all' failed
> make[1]: *** [tools/VideoViewer/CMakeFiles/VideoViewer.dir/all] Error 2
> Makefile:151: recipe for target 'all' failed
> make: *** [all] Error 2
> ```
> 
> When I run the first cmake, I got many warnings of the following type:
> 
> ```
> CMake Warning at tools/VideoViewer/CMakeLists.txt:5 (add_executable):
>   Cannot generate a safe runtime search path for target VideoViewer because
>   files in some directories may conflict with libraries in implicit
>   directories:
> 
>     runtime library [libz.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
>       /home/kovan-fl/anaconda3/lib
>     runtime library [libtiff.so.5] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
>       /home/kovan-fl/anaconda3/lib
>     runtime library [liblz4.so.1] in /usr/lib/x86_64-linux-gnu may be hidden by files in:
>       /home/kovan-fl/anaconda3/lib
> 
>   Some of these libraries may not be found correctly.
> ```
> 
> Then, I deleted the old build and tried to build without anaconda, and it worked. Also, the dependencies are installed via `sudo apt-get install`, not with `conda install`; therefore, I believe, under normal circumstances, you should not see an error about dependencies.
> 
> Regards,
> 
> Burak

Thanks sincerely, I've solved this problem a months ago, and I built it without anaconda too.

**Comment by rancheng** [comment]
I solved this issue by compile zlib 1.2.9 and copy into miniconda lib folder:

credit to [StackOverFlow](https://stackoverflow.com/questions/48306849/lib-x86-64-linux-gnu-libz-so-1-version-zlib-1-2-9-not-found)

[Download](https://sourceforge.net/projects/libpng/files/zlib/1.2.9/zlib-1.2.9.tar.gz/download) zlib 1.2.9.

```sh 
tar -xvf ~/Downloads/zlib-1.2.9.tar.gz
cd zlib-1.2.9
sudo -s
./configure; make; make install
```
then copy your `libz.a` and `libz.so.1.2.9` into your miniconda library folder:

```sh
cp /usr/local/lib/libz.a /home/XXX/miniconda3/lib/
cp /usr/local/lib/libz.so.1.2.9 /home/XXX/miniconda3/lib/
``` 

build again, should be sucessful now.
